packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx

```diff
@@ -303,25 +305,53 @@ const OrderDetailsTable: React.FC<OrderDetailsProps> = ({ patientUuid, showAddBu
     [orderTypes, t],
   );
 
+  const handleDateFilterChange = ([startDate, endDate]) => {
+    if (startDate) {
+      setSelectedFromDate(startDate.toISOString());
+      if (selectedToDate && startDate && selectedToDate < startDate) {
```

Comment:
```suggestion
      if (selectedToDate && selectedToDate < startDate) {
```

Isn't the outer check sufficient?